anda/langs/crystal/crystal/crystal.spec

```diff
@@ -1,57 +1,49 @@
 %define debug_package %nil
 
-Name:			crystal
+Name:			  crystal
 Version:		1.18.2
-Release:		1%?dist
-Summary:		The Crystal Programming Language
+Release:		2%?dist
+Summary:    A general-purpose, object-oriented programming language
 License:		Apache-2.0
-URL:			https://crystal-lang.org/
-Source0:		https://github.com/crystal-lang/crystal/releases/download/%version/crystal-%version-1-linux-x86_64-bundled.tar.gz
-ExclusiveArch:	x86_64
-BuildRequires:	rpm_macro(fdupes)
+URL:			  https://crystal-lang.org/
+Source0:    https://github.com/crystal-lang/crystal/archive/%version.tar.gz
+BuildRequires:  crystal gcc gcc-c++ make gc-devel llvm-devel
+BuildRequires:  pcre2-devel libyaml-devel libffi-devel
+Requires:       gcc pkgconfig gc-devel
+Requires:       pcre2-devel openssl-devel zlib-devel
+Requires:       libyaml-devel libxml2-devel gmp-devel
+Suggests:       shards
 
 %description
-%summary.
-
-%package devel
-Summary:	Development files for the crystal package
-
-%description devel
-%summary.
+Crystal is a programming language with the following goals:
+- Have a syntax similar to Ruby (but compatibility with it is not a goal)
+- Statically type-checked but without having to specify the type of variables or method arguments
+- Be able to call C code by writing bindings to it in Crystal
+- Have compile-time evaluation and generation of code, to avoid boilerplate code
+- Compile to efficient native code
 
 %prep
-%autosetup -n crystal-%version-1
+%setup -q
 
 %build
+make release=1 interpreter=1 LDFLAGS="%{build_ldflags}" CRYSTAL_CONFIG_LIBRARY_PATH=%{_libdir}/crystal
 
 %install
-mkdir -p %buildroot/usr/bin
-mkdir -p %buildroot/usr/share
-mkdir -p %buildroot/usr/lib/crystal
-install -Dm755 bin/* %buildroot/usr/bin/
-cp -r share/* %buildroot/usr/share/
-cp -r lib/crystal/* %buildroot/usr/lib/crystal/
-
-%fdupes %buildroot%_datadir/crystal/src/lib_c/
-
+make install DESTDIR=%{buildroot} PREFIX=%{_prefix}
 
 %files
-%license /usr/share/licenses/crystal/LICENSE
-/usr/bin/crystal
-/usr/share/zsh/site-functions/_crystal
-/usr/share/man/man1/crystal.1.gz
-/usr/share/crystal/
-/usr/share/fish/vendor_completions.d/crystal.fish
-/usr/share/bash-completion/completions/crystal
-
-/usr/bin/shards
-/usr/share/man/man1/shards.1.gz
-/usr/share/man/man5/shard.yml.5.gz
-
-%files devel
-/usr/lib/crystal/
+%license %{_datadir}/licenses/crystal/LICENSE
+%{_bindir}/crystal
+%{_datadir}/crystal
+%{_datadir}/zsh/site-functions/_crystal
+%{_datadir}/bash-completion/completions/crystal
+%{_datadir}/fish/vendor_completions.d/crystal.fish
```

Comment:
We haven't updated this spec file in a while, but we have new guidelines and macros now.
We can remove these 3 lines and put `%pkg_completion -Bfz` before `%prep`

Reply:
not possible before `%description`. between `%description` and `%prep` yielded this error